### PR TITLE
(#22) add jest for testing with example test

### PR DESCRIPTION
**Pull request by davidzwa**
Adds respective GHA workflow with CodeCov. Internally adds the packages for testing with mongo, jest for tests and jest-utils for common utils needed later.

## Description

Add testing!

## Related Issue

#22 

## Motivation and Context

This solves devs being able to write automated tests with coverage reports.

## How Has This Been Tested?

How the tests have been tested? Goes a bit far ;)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/nodejs-pr.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - run: yarn install --frozen-lockfile
+      - run: yarn run test:cov
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
```
Comment (mkevenaar) [inline .github/workflows/nodejs-pr.yaml:26]:
```suggestion
        uses: codecov/codecov-action@v3.1.0
```

File: src/bot.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const jsExt = Constants.jsExt;
  * Creates the discord client with interface hooks
  * @returns {Client<boolean>}
  */
-function createDiscordClient() {
+export function createDiscordClient() {
```
Comment (mkevenaar) [inline src/bot.js:18]: Why is this here? Is this a bugfix discovered by testing?

**Review by mkevenaar — CHANGES_REQUESTED** [review]
Just some small comments, other than that, LGTM!

File: src/bot.js (hunk 2 above)
Comment (Lucxjo) [reply src/bot.js:18]: The `test/application/bot.test.js` needs access to this function looking at the code so this should be fine

**Review by mkevenaar — APPROVED** [review]

**Comment by mkevenaar** [comment]
@all-contributors please add @davidzwa for test

**Comment by mkevenaar** [comment]
@all-contributors please add @davidzwa for review

**Comment by mkevenaar** [comment]
@all-contributors please add @Lucxjo for review

**Comment by mkevenaar** [comment]
https://github.com/all-contributors please add @Lucxjo for review

**Comment by mkevenaar** [comment]
@all-contributors please add @Lucxjo for review

**Comment by mkevenaar** [comment]
@all-contributors check
